src/platform/plugins/shared/visualizations/common/convert_to_lens/types.ts

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { $Values } from '@kbn/utility-types';
+import type {
+  AvgIndexPatternColumn,
+  CardinalityIndexPatternColumn,
+  CountIndexPatternColumn,
+  CounterRateIndexPatternColumn,
+  CumulativeSumIndexPatternColumn,
+  DataType,
+  DateHistogramIndexPatternColumn,
+  DerivativeIndexPatternColumn,
+  FiltersIndexPatternColumn,
+  FormulaIndexPatternColumn,
+  GenericIndexPatternColumn,
+  LastValueIndexPatternColumn,
+  MaxIndexPatternColumn,
+  MedianIndexPatternColumn,
+  MinIndexPatternColumn,
+  MovingAverageIndexPatternColumn,
+  PercentileIndexPatternColumn,
+  PercentileRanksIndexPatternColumn,
+  RangeIndexPatternColumn,
+  StandardDeviationIndexPatternColumn,
+  StaticValueIndexPatternColumn,
+  SumIndexPatternColumn,
+  TermsIndexPatternColumn,
+  TimeScaleUnit,
+  ValueFormatConfig,
+} from '@kbn/lens-common';
+import type { Query } from '@kbn/es-query';
+import type { Operations, OperationsWithSourceField, OperationsWithReferences } from './constants';
+
+export interface MinMax {
+  min: number;
+  max: number;
+}
+
+export interface BasicFullPercentageModeConfig {
+  isPercentageMode: boolean;
+}
+
+export interface BasicPercentageModeConfig {
+  isPercentageMode: false;
+}
+
+export type PercentageModeConfigWithMinMax = BasicFullPercentageModeConfig & MinMax;
+
+export type PercentageModeConfig = BasicPercentageModeConfig | PercentageModeConfigWithMinMax;
+
+export interface BaseColumn<OperationType extends Operation, Params = undefined> {
+  columnId: string;
+  operationType: OperationType;
+  label?: string;
+  isBucketed: boolean;
+  isSplit: boolean;
+  dataType: DataType;
+  timeScale?: TimeScaleUnit; // ?
```

Comment:
missing something here?

Reply:
I am not aware of the original reason for that comment.
The type was copied as is from the original source: https://github.com/elastic/kibana/pull/239399/files/f18554a6c7a1a79f3cf93a687e6362decaef6264#diff-bc02f035c61a2f4962b7c2139443b1f22bacb4fc09c87b99aac40020ddea7048L43